### PR TITLE
Upgrade Flutter used for build, and fix linter warnings

### PR DESCRIPTION
**Pull request by stevehobbsdev**
- N/A All new/changed/fixed functionality is covered by tests (or N/A)
- N/A I have added documentation for all new/changed functionality (or N/A)

### 📋 Changes

- Upgrade version of Flutter used in build to 3.7.4
- Fix linter warnings, mostly around lines > 80 chars in comments. A couple of tests violated this, thought it better to ignore the file rather than obfusticate the test name for this

Primary motivation was that Ship CLI fails when running `flutter analyze` (now returns a non-zero exit code even for warnings). Plus, good to keep up to date with latest Flutter anyway, and linter warnings should be fixed before they get out of hand.

### CODE REVIEW

File: auth0_flutter/analysis_options.yaml [hunk 1]
```diff
@@ -52,7 +52,7 @@ linter:
     join_return_with_assignment: true
     leading_newlines_in_multiline_strings: true
     library_private_types_in_public_api: true
-    lines_longer_than_80_chars -> packages only: true
```
Comment (stevehobbsdev) [inline auth0_flutter/analysis_options.yaml:55]: Flutter CLI doesn't know what `packages only` does any more, and I can't find much on the web about it.

File: auth0_flutter/example/lib/api_card.dart [hunk 2]
```diff
@@ -6,7 +6,7 @@ import 'constants.dart';
 
 class ApiCard extends StatefulWidget {
   final Future<void> Function(String usernameOrEmail, String password) action;
-  const ApiCard({required final this.action, final Key? key}) : super(key: key);
```
Comment (stevehobbsdev) [inline auth0_flutter/example/lib/api_card.dart:9]: Fixes `unnecessary final` warnings, as these are implicitly `final`.

File: auth0_flutter/analysis_options.yaml (hunk 1 above)
Comment (Widcket) [reply auth0_flutter/analysis_options.yaml:55]: That's a comment by me, which I forgot to remove when I uncommented the line. My bad 🤦🏼‍♀️

File: auth0_flutter/example/analysis_options.yaml [hunk 3]
```diff
@@ -73,7 +73,7 @@ linter:
     join_return_with_assignment: true
     leading_newlines_in_multiline_strings: true
     library_private_types_in_public_api: true
-    lines_longer_than_80_chars -> packages only: true
+    lines_longer_than_80_chars: true
```
Comment (Widcket) [inline auth0_flutter/example/analysis_options.yaml:76]: Could you please change `todo` to `warning` on L29? Also for the  other YAML files.

File: auth0_flutter/analysis_options.yaml (hunk 1 above)
Comment (stevehobbsdev) [reply auth0_flutter/analysis_options.yaml:55]: All good, I had thought it was some kind of special hidden syntax we were using 😄

File: auth0_flutter/example/analysis_options.yaml (hunk 3 above)
Comment (stevehobbsdev) [reply auth0_flutter/example/analysis_options.yaml:76]: Done in 110bd0c

**Review by Widcket — APPROVED** [review]
Thank you!

**Pull request merged**